File: 3457.eat-pizzas.py
#
# @lc app=leetcode id=3457 lang=python3
#
# [3457] Eat Pizzas!
#

# @lc code=start
from typing import List

class Solution:
    def maxWeight(self, pizzas: List[int]) -> int:
        pizzas.sort(reverse=True)
        n = len(pizzas)
        k = n // 4
        num_odd = (k + 1) // 2
        num_even = k // 2
        total = sum(pizzas[:num_odd])
        pos = num_odd
        for i in range(num_even):
            total += pizzas[pos + 2 * i + 1]
        return total

# @lc code=end

File: 3457.eat-pizzas.cpp
#
# @lc app=leetcode id=3457 lang=cpp
#
# [3457] Eat Pizzas!
#
# @lc code=start
class Solution {
public:
    long long maxWeight(vector<int>& pizzas) {
        int n = pizzas.size();
        sort(pizzas.begin(), pizzas.end(), greater<int>());
        
        int total_days = n / 4;
        int odd_days = (total_days + 1) / 2;
        int even_days = total_days / 2;
        
        long long result = 0;
        
        // Sum top odd_days pizzas (gained on odd days)
        for (int i = 0; i < odd_days; i++) {
            result += pizzas[i];
        }
        
        // Sum every second pizza from the next 2*even_days pizzas
        // (second-max of each even-day group)
        for (int k = 1; k <= even_days; k++) {
            int idx = odd_days + 2 * k - 1;
            result += pizzas[idx];
        }
        
        return result;
    }
};
# @lc code=end